api/v1beta1/azurecluster_default.go

```diff
@@ -124,10 +124,14 @@ func (c *AzureCluster) setSubnetDefaults() {
 		if subnet.RouteTable.Name == "" {
 			subnet.RouteTable.Name = generateNodeRouteTableName(c.ObjectMeta.Name)
 		}
-		if subnet.IsNatGatewayEnabled() {
-			if subnet.NatGateway.NatGatewayIP.Name == "" {
-				subnet.NatGateway.NatGatewayIP.Name = generateNatGatewayIPName(c.ObjectMeta.Name, subnet.Name)
+
+		if !subnet.IsIPv6Enabled() {
+			// NAT gateway supports the use of IPv4 public IP addresses for outbound connectivity.
+			// So default use the NAT gateway for outbound traffic in IPv4 cluster instead of loadbalancer.
+			if !subnet.IsNatGatewayEnabled() {
```

Comment:
this could just be 

`if subnet.NatGateway.Name == ""` 

the `IsNatGatewayEnabled` can probably be removed, or just changed to 

```
func (s SubnetSpec) IsNatGatewayEnabled() bool {
  return !s.IsIPv6Enabled()
}

Reply:
@CecileRobertMichon I change it ti be `if subnet.NatGateway.Name == ""`.